crypto/params.c

```diff
@@ -16,8 +16,10 @@ OSSL_PARAM *OSSL_PARAM_locate(OSSL_PARAM *p, const char *key)
 {
     if (p != NULL && key != NULL)
         for (; p->key != NULL; p++)
-            if (strcmp(key, p->key) == 0)
+            if (strcmp(key, p->key) == 0) {
+                p->flags |= OSSL_PARAM_USED;
```

Comment:
you need a set-used method for apps that don't call locate.  replace set-unused with this:
```
void OSSL_PARAM_set_used(OSSL_PARAM *p. int used);
```

Reply:
This was the alternative approach.  I'm unsure which is better.

Reply:
I've switch to this approach.